opentelemetry/proto/collector/dynamicconfig/v1/dynamicconfig_service.proto

```diff
@@ -0,0 +1,159 @@
+// having to restart the instrumented application. The collector may also act as
+// a bridge between third-party configuration services and the agent/sdk, piping
+// updated configs from a third-party source to an instrumented application.
+service DynamicConfig {
+  rpc GetConfig (ConfigRequest) returns (ConfigResponse);
+}
+
+message ConfigRequest{
+  // Describes the version of the config service protocol to use. Required.
+  int32 proto_version = 1;
+
+  // The target resource. This resource pings the config server requesting
+  // updated configs, and the server responds with the config information.
+  opentelemetry.proto.resource.v1.Resource resource = 2;
+
+  // Optional. Identifies the last configuration fingerprint / version number.
+  // The fingerprint serves as a quick way for the config server to determine
+  // whether an update to the configs had occurred -- if the client request's
+  // fingerprint differs from the server's, then the server knows that a change
+  // had occurred and an updated configuration should be sent.
+  string last_fingerprint = 3;
+}
+
+message ConfigResponse {
+  // Describes the version of the config service protocol to use. Should match
+  // the request’s version.
+  int32 proto_version = 1;
+
+  // Optional. The fingerprint associated with the current config settings. Each
+  // change in configs yields a different fingerprint. The client may cache a
+  // fingerprint, then return it on the next request to enable the server to
+  // quickly check whether a change in the configs occurred. If no changes
+  // occurred, then all other fields in the response are optional.
+  string fingerprint = 2;
+
+  // Dynamic configs specific to metrics, namely the metric collection schedule
+  message MetricConfig {
+
+    // A Schedule is used to apply a particular scheduling configuration to
+    // a metric. If a metric name matches a schedule's patterns, then the metric
+    // adopts the configuration specified by the schedule.
+    message Schedule {
+
+      // Pattern dictates a light-weight pattern that can match 1 or more
+      // metrics, for which this schedule will apply. The string is used to
+      // match against metric names. It should not exceed 100k characters.
+      message Pattern {
+        oneof match {
+          string equals = 1;       // matches the metric name exactly
+          string starts_with = 2;  // prefix-matches the metric name
+        }
+      }
+
+      // Metrics with names that match a rule in the inclusion_patterns are
+      // targeted by this schedule. Metrics that match the exclusion_patterns
+      // are not targeted for this schedule, even if they match an inclusion
+      // pattern.
+      repeated Pattern inclusion_patterns = 1;
+      repeated Pattern exclusion_patterns = 2;
+
+      // CollectionPeriod describes the sampling period for each metric. All
+      // larger units are divisible by all smaller ones.
```

Comment:
The original intent was to enforce the property that all larger units be divisible by smaller ones (which makes the implementation much cleaner), though we're thinking of altering this -- for example in case a vendor has an extremely good reason to use a collection period of, say, 2 days